### ISSUE TITLE
ElasticFields.Score Query

### ISSUE THREAD

**Issue by wilsoncgp** [ElasticLINQ/ElasticLINQ#52]
Hi there,

I'm trying to create a full text query on my data and because it uses query_string, it is returning all data.

I only want the relevant results, so I want to use the ElasticFields.Score field in my queries to get all results with a score greater than 1.0, yet I can't seem to find any info on using it barring a near 2 year old video where the field gets mentioned, it doesn't work for the speaker, then he moves onto something else.

Similarly to myself, if I try to use the field, it makes my query return no results, it should return at least 1 result as the query before it contains 3 results, 1 of which has a score greater than 1.0. I'm using a standard Linq expression predicate to pass through to a function so my code looks like this:

```
public IQueryable<MyClassName> SearchFor(System.Linq.Expressions.Expression<Func<MyClassName, bool>> predicate)
{
    return context.Query<MyClassName>().Query(predicate);
}
```

This returns the results as planned. After adding in the ElasticFields.Score field:

```
public IQueryable<MyClassName> SearchFor(System.Linq.Expressions.Expression<Func<MyClassName, bool>> predicate)
{
    return context.Query<MyClassName>().Query(predicate).Where(p => ElasticFields.Score > 1);
}
```

Which returns no results. Do you know if there is an issue with the ElasticField.Score field?

The query that comes out of the above is the following:

<pre>
{
  "query":{
    "query_string":{
      "query": "*{Query String}*",
      "fields": ["{Field to query on}"]
    }
  },
  "filter":{
    "range":{
      "_score":{
        "gt":1.0
      }
    }
  },
  "timeout":"10s"
}
</pre>

However, it seems that elasticsearch is expecting:

<pre>
{
  "min_score": "1.0",
  "query":{
    "query_string":{
      "query": "*{Query String}*",
      "fields": ["{Field to query on}"]
    }
  },
  "timeout":"10s"
}
</pre>

Thanks,

Liam.

**Comment by damieng** [comment]
Hi Liam.

ElasticLINQ does not yet support min_score (the fact it generates a silly filter is a bug). It's not difficult to add but hadn't come in as a request from anyone. I might be able to get something checked in over the next 24 hours if that helps you.

**Comment by damieng** [comment]
I've checked in an experimental version into dev that adds a MinScore method to the query chain, e.g.

``` csharp
context.Query<MyClassName>().Query(predicate).MinScore(2)
```

I'm considering the ElasticFields.Score property but there are a few issues there including:
1. We'd only support >= not < > <= as Elasticsearch only has min_score which is inclusive
2. Should probably only be in .Query not .Where as latter is for filtering
3. Would need to only support it as a top-level or an AND at top level

So the following would probably be the only valid syntax for using ElasticFields.Score:

``` csharp
context.Query<MyClassName>().Query(predicate && ElasticFields.Score >= 2);
```

**Comment by wilsoncgp** [comment]
Indeed, maybe the ElasticFields.Score property being used isn't particularly relevant given that it would only be able to be used with the '>=' operator. The MinScore method would definitely be useful for me.

**Comment by bgeihsgt** [comment]
+1 to the MinScore approach over supporting the raw field.  Good to create a pit of success instead of a pit of failure.

**Comment by damieng** [comment]
Shipped in 1.3.1
